### DISCUSSION TITLE
Official Docker Image

### DISCUSSION THREAD

**Discussion by winenerd-ajr** [prisma/prisma#2567] [Q&A] [answered]
## Problem
Currently, I don't see any official Docker images for prisma (2). This means everyone has to roll their own Dockerfiles, and considering the fact that this is targeted as a project to help developers interact with databases more quickly (and, for newer developers, more easily), it does not make sense that developers should have to do that on their own with all the potential pitfalls that can occur. The vast majority of services written today provide official Docker images to prevent misconfiguration. Of course, this is even more beneficial for production environments than local development (though a well-supported docker image would make local development easier as well), but if you're not planning to deploy this in production, why run it?

## Suggested solution
1. Upload an official (Alpine-based) docker image to docker hub.
2. Write a guide for development using docker-compose and interactive TTYs.
3. Write a guide for deploying to Kubernetes.

## Alternatives
Don't do this, and push people to use Hasura (which is what I'm considering doing instead of migrating from Prisma 1 -- if there's a nonstandard SDL and no simple deployment, why not just use something that is similarly kneecapped but easier to use out of the box?).

## Additional context
This should have been an MVP feature imho.

**Comment by Sytten** [comment, marked as answer]
Prisma 2 is very different from Prisma 1, its not a separate server anymore. So I don't really know what you would use a Prisma image for. You can take a look at the guide I posted: https://medium.com/@emilefugulin/building-a-sane-docker-image-for-typescript-lerna-and-prisma-2-76d8ff9926e4
As for deploying to kubernetes, it is pretty much the same as any service you would deploy, its just the docker image of your service with an environment variable to set the database URL.
Honestly Hasura is a very different tool and not aimed at the same audience.

**Comment by pantharshit00** [comment]
Yeah, Prisma is now a node.js library. You will treat your app as any other node js project. No special Prisma server deployment needed now.

**Comment by ludralph** [comment]
Hi @winenerd-ajr,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
